fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -408,99 +393,61 @@ public Response createOrReplaceObjectRdf(
-
-            if (httpConfiguration.putRequiresIfMatch() && StringUtils.isBlank(ifMatch) && !resource.isNew()) {
-                throw new ClientErrorException("An If-Match header is required", 428);
+                createResourceService.perform(transaction.getId(), fedoraId, null, false, contentType,
+                        links, checksums, requestBodyStream, extContent);
             }
+        } else {
+            final Model model = httpRdfService.bodyToInternalModel(externalPath(), requestBodyStream,
+                requestContentType);
 
-            evaluateRequestPreconditions(request, servletResponse, resource, transaction);
-            final boolean created = resource.isNew();
-
-            try (final RdfStream resourceTriples =
-                    created ? new DefaultRdfStream(asNode(resource())) : getResourceTriples(resource())) {
-                if (resource instanceof Binary) {
-                    InputStream stream = requestBodyStream;
-                    MediaType type = requestContentType;
-                    // override a few things, if it's external content
-                    if (extContent != null) {
-                        if (extContent.isCopy()) {
-                            LOGGER.debug("External content COPY '{}', '{}'", externalPath, extContent.getURL());
-                            stream = extContent.fetchExternalContent();
-                        }
-
-                        type = contentType;  // if external, then this already holds the correct value
-                    }
-                    final String handling = extContent != null ? extContent.getHandling() : null;
-                    replaceResourceBinaryWithStream((Binary) resource,
-                            stream, contentDisposition, type, checksums,
-                            (handling != null && !handling.equals(COPY)) ? handling : null,
-                            (extContent != null && !handling.equals(COPY)) ? extContent.getURL() : null);
-
-                } else if (isRdfContentType(contentType.toString())) {
-                    replaceResourceWithStream(resource, requestBodyStream, contentType, resourceTriples);
-                } else if (!created) {
-                    boolean emptyRequest = true;
-                    try {
-                        emptyRequest = requestBodyStream.read() == -1;
-                    } catch (final IOException ex) {
-                        LOGGER.debug("Error checking for request body content", ex);
-                    }
-
-                    if (requestContentType == null && emptyRequest) {
-                        throw new ClientErrorException("Resource Already Exists", CONFLICT);
-                    }
-                    throw new NotSupportedException("Invalid Content Type " + requestContentType);
-                }
-            } catch (final Exception e) {
-                checkForInsufficientStorageException(e, e);
+            if ("Resource Exists".isEmpty()) {
```

Comment:
I'm just realize that this pseudo code sounds wrong. Not sure what I was thinking but it meant the object does exist.